python/ray/data/tests/test_parquet.py

```diff
@@ -824,6 +824,57 @@ def test_parquet_write_create_dir(
     else:
         fs.delete_dir(_unwrap_protocol(path))
 
+    # Test that writing empty blocks does not create empty parquet files,
+    # nor does it create empty directories when no files are created.
+    ds_all_empty = ds.filter(lambda x: x["one"] > 10).materialize()
+    assert ds_all_empty.num_blocks() == 2
+    assert ds_all_empty.count() == 0
+
+    all_empty_key = "all_empty"
+    all_empty_path = os.path.join(data_path, f"test_parquet_dir_{all_empty_key}")
+    ds_all_empty._set_uuid(all_empty_key)
```

Comment:
Why do we need to call `_set_uuid`?

Reply:
https://github.com/ray-project/ray/pull/36134#discussion_r1222382259